source/web/js_utils.h

```diff
@@ -580,7 +580,96 @@ namespace emp {
     }
   }
 
-/// @endcond
+  /// @endcond
+
+  /// This function can be called to pass a map into JavaScript.
+  /// The resulting JavaScript object will be stored in emp.__incoming_map. 
+  /// @param dict the map being passed into JavaScript
+  template <typename KEY_T, typename VAL_T>
+  void pass_map_to_javascript(const std::map<KEY_T, VAL_T> & dict) {
+   
+    emp::vector<KEY_T> keys;
+    emp::vector<VAL_T> values;
+
+    // extract keys and values from dict
+    for (typename std::map<KEY_T, VAL_T>::const_iterator it = dict.begin(); it != dict.end(); ++it) {
+      keys.push_back(it->first);
+      values.push_back(it->second);
+    }
+
+    // pass in extracted keys vector to JS
+    emp::pass_array_to_javascript(keys);
+    EM_ASM({
+      emp_i.__incoming_map_keys = emp_i.__incoming_array;
+    });
+
+    // check to make sure each key is not an object or a function
+    emp_assert(
+        EM_ASM_INT({
+          emp_i.__incoming_map_keys.forEach(function(key) {
+            if (typeof key === "object" || typeof key === "function") { return 0; }
+          });
+          return 1;
+        }), "Keys cannot be an object or a function");
+
+    // pass in extracted values vector to JS
+    emp::pass_array_to_javascript(values);
+    EM_ASM({
+      emp_i.__incoming_map_values = emp_i.__incoming_array;
+
+      // create dictionary
+      emp_i.__incoming_map = ( {} );
+      emp_i.__incoming_map_keys.forEach(function(key, val) {
+        emp_i.__incoming_map[key] = emp_i.__incoming_map_values[val]
+      });
+
+      // clean up unneeded vars
+      delete emp_i.__incoming_map_keys;
+      delete emp_i.__incoming_map_values;
+    });
+  }
+
+ 
+  /// This function can be called to pass two arrays of the same length into JavaScript (where a map is then created) 
+  /// One array should hold keys, and the other should hold values 
+  /// (note that the key-value pairs must line up across the arrays)
+  /// The resulting JavaScript object will be stored in emp.__incoming_map. 
+  /// @param keys an array holding the keys to the map
+  /// @param values an array holding the values to the map
+  template <typename KEY_T, typename VAL_T, size_t SIZE>
+  void pass_map_to_javascript(const emp::array<KEY_T, SIZE> & keys, const emp::array<VAL_T, SIZE> & values) {
```

Comment:
I didn't end up doing this since when I extract the keys and values from an `emp::map` I need to store those keys and values as vectors, meaning I can't pass it into the `pass_map_to_javascript` function that takes arrays (and I think it makes sense to keep these as arrays in order to make sure that each key has an associated value).